numpy/lib/_function_base_impl.py

```diff
@@ -1867,62 +1867,130 @@ def sort_complex(a):
         return b
 
 
-def _trim_zeros(filt, trim=None):
+def _arg_trim_zeros(filt):
+    """Return indices of the first and last non-zero element.
+
+    Parameters
+    ----------
+    filt : array_like
+        Input array.
+
+    Returns
+    -------
+    start, stop : ndarray
+        Two arrays containing the indices of the first and last non-zero
+        element in each dimension.
+
+    See also
+    --------
+    trim_zeros
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> _arg_trim_zeros(np.array([0, 0, 1, 1, 0]))
+    (array([2]), array([3]))
+    """
+    nonzero = np.argwhere(filt != 0)
```

Comment:
Had to use `filt != 0` here instead of just passing `filt` because `np.argwhere` (and `nonzero`) treat `None` as zero. This behavior is also covered by an already existing test [`test_no_trim`](https://github.com/numpy/numpy/blob/cecc005ff2b2ac133f0f535fa86c781dfcb9d1ae/numpy/lib/tests/test_function_base.py#L1391-L1394).

I'm a bit surprised by this behavior. Not sure if there's a faster way than adding the extra `filt != 0` check.

Reply:
Only do this if it's an object array.